apps/meteor/app/lib/server/functions/saveUser.ts

```diff
@@ -398,23 +394,15 @@ export const saveUser = async function (userId, userData) {
 		sendPassword = false;
 	}
 
-	const updateUser = {
-		$set: {},
-	};
+	const updateUser: DeepWritable<UpdateFilter<IUser>> = {};
+	updateUser.$set = {};
 
 	handleBio(updateUser, userData.bio);
 	handleNickname(updateUser, userData.nickname);
 
 	if (userData.roles) {
 		updateUser.$set.roles = userData.roles;
 	}
-	if (userData.settings) {
-		updateUser.$set.settings = { preferences: userData.settings.preferences };
-	}
-
-	if (userData.language) {
-		updateUser.$set.language = userData.language;
-	}
```

Comment:
These fields don't exist in the `IUpdateUserParams` type (from `userData`). Keeping them would cause a typecheck error -- basically this should not be there since we don't even use it

Reply:
But we are using the same `saveUser` function in multiple places and one of them is the `insertOrUpdateUser` meteor method and we are not validating anything there. If we just remove this, it should be considered a breaking change 😞

Reply:
Ohhhh, I see. So I think this conversion will require us to add these params to the `rest-typings` definition too. Sounds ok to you? :grimacing: